packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts

```diff
@@ -921,6 +1157,79 @@ describe('plugin-dss', () => {
         expect(Batcher.prototype.request.getCall(1).args).to.deep.equal(['id2']);
         expect(result).to.equal(response2);
       });
+
+      it('fails when mercury does not response but only the affected batch', async () => {
+        const requests = [
+          {
+            id: 'req-id-1',
+            resource: '/lookup/orgid/userOrgId/identities',
+            bodyParams: {lookupValues: ['id1', 'id2', 'id3']},
+          },
+        ];
+
+        const {
+          promises: [p1, p2, p3],
+        } = await testMakeBatchedRequests({
+          requests,
+          calls: [
+            {
+              method: 'lookup',
+              params: {id: 'id1', shouldBatch: true},
+            },
+            {
+              method: 'lookup',
+              params: {id: 'id2', shouldBatch: true},
+            },
+            {
+              method: 'lookup',
+              params: {id: 'id3', shouldBatch: true},
+            },
+          ],
+        });
+
+        // only 1 response
+        mercuryCallbacks['event:directory.lookup'](
+          createData('id 1', 0, true, 'lookupResult', {entitiesNotFound: ['id1', 'id13']})
+        );
+
+        // Timeout
+        await clock.tickAsync(6000);
+
+        // Next batch
+        uuidStub.onCall(requests.length).returns('req-id-2');
```

Comment:
why not use testMakeBatchedRequests again?

Reply:
fixed